### PR TITLE
tweaks data model

### PR DESCRIPTION
**Pull request by mlodic**
@cristinaascari please review comments

### CODE REVIEW

File: api_app/analyzers_manager/models.py [hunk 1]
```diff
@@ -143,6 +145,7 @@ def create_data_model(self) -> Optional[BaseDataModel]:
         data_model = self.data_model_class.objects.create(**dictionary)
         self.data_model = data_model
         self.save()
+        # todo so for every analysis we do create a new data model element in the db. And how the retention is managed?
```
[the remark this answers is not in the document]
Comment (cristinaascari) [reply api_app/analyzers_manager/models.py:148]: This has already been fixed in the datamodel_visualizer branch. If the analyzer doesn't have a mapping in its config and it has not overwritten the methods `_create_data_model_mtm` and `_update_data_model` of the `BaseAnalyzerMixin` class, the data model will not be created

[the remark this answers is not in the document]
Comment (mlodic) [reply api_app/analyzers_manager/models.py:148]: great! just one thing: fixes should be separated if possible from other developments and have separate commits.
In this way it is more easy to manage branches and have a better history of what happened. Plus, a possible rollback would affect only the fix itself and not the rest of the code. So isolation should be prioritized for fixes

File: api_app/analyzers_manager/models.py [hunk 2]
```diff
@@ -106,14 +107,15 @@ def _create_data_model_dictionary(self) -> Dict:
...
                     value = self.get_value(self.report, report_key.split("."))
-                    logger.info(f"Retrieved {value} from key {report_key}")
+                    logger.debug(f"Retrieved {value} from key {report_key}")
                 except Exception:
                     # validation
-                    self.errors.append(f"Field {report_key} not present in report")
+                    self.errors.append(f"Field {report_key} not available in report")
                     continue
-                    # create the related object if necessary
+                    # create the related object if necessary #todo what does it mean - there is this comment then nothing
```
[the remark this answers is not in the document]
Comment (0ssigeno) [reply api_app/analyzers_manager/models.py:115]: is referencing this code 
```            
if isinstance(data_model_fields[data_model_key], ForeignKey):
                # to create an object we need at least
                if not isinstance(value, dict):
                    self.errors.append(
                        f"Field {report_key} has type {type(report_key)} while a dictionary is expected"
                    )
                    continue
                value, _ = data_model_fields[
                    data_model_key
                ].related_model.objects.get_or_create(**value)
                result[data_model_key] = value
```
that is after the comment

File: api_app/analyzers_manager/models.py [hunk 3]
```diff
@@ -106,14 +107,15 @@ def _create_data_model_dictionary(self) -> Dict:
...
+
             if isinstance(data_model_fields[data_model_key], ForeignKey):
-                # to create an object we need at least
+                # to create an object we need at least #todo what does it mean - sentence not finished
```
[the remark this answers is not in the document]
Comment (0ssigeno) [reply api_app/analyzers_manager/models.py:118]: _at least a dictionary_

```
 not isinstance(value, dict):
 ```

**Review by cristinaascari — APPROVED** [review]

**Review by mlodic — COMMENTED** [review]
approved for me, we can go forward

**Comment by drosetti** [comment]
Note: some tests are failing, but they are related to the dashboard not to this feature, i'll fix them asap

**Pull request merged**